### PR TITLE
Update comment and fix test for Marshal.GetDelegateForFunctionPointer argument validation

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 1]
```diff
@@ -867,15 +867,6 @@ public static Delegate GetDelegateForFunctionPointer(IntPtr ptr, Type t)
                 throw new ArgumentException(SR.Argument_NeedNonGenericType, nameof(t));
             }
 
-            // COMPAT: This block of code isn't entirely correct.
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:870]: This check is redundant with a check done on the unmanaged side already.

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs [hunk 2]
```diff
@@ -75,6 +76,17 @@ public void GetDelegateForFunctionPointer_GenericInvalidType_ReturnsExpected()
             VerifyDelegate(functionDelegate, targetMethod);
         }
 
+        [Fact]
+        public void GetDelegateForFunctionPointer_DelegateBaseTypes_ThrowsMustBeDelegate()
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs:80]: Added test coverage for the path that I am touching. I expect that this test will fail on Mono (will file bug).

**Comment by GrabYourPitchforks** [comment]
@jkotas Did you mean to close this without merging? The changes LGTM.

**Comment by jkotas** [comment]
I have realized that the change is not good as is. It allowed `Marshal.GetFunctionPointerForDelegate(..., typeof(string))` to succeed in some cases. I will reopen once I fix it up.

**Pull request closed without merging**